project_task_stage_state/models/project_task_type.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2014 Daniel Reis
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import fields, models
+
+_TASK_STATE = [
+    ("draft", "New"),
+    ("open", "In Progress"),
+    ("pending", "Pending"),
+    ("done", "Done"),
+    ("cancelled", "Cancelled"),
+]
+
+
+class ProjectTaskType(models.Model):
+    """Added state in the Project Task Type."""
+
+    _inherit = "project.task.type"
+
+    state = fields.Selection(_TASK_STATE)
```

Comment:
Maybe the best approach to allow reusability of the module will be:
```
@api.model
def _get_task_states(self):
    return [
        ("draft", "New"),
        ("open", "In Progress"),
        ("pending", "Pending"),
        ("done", "Done"),
        ("cancelled", "Cancelled"),
    ]

state = fields.Selection(selection="_get_task_states")
```

It will increase reusability because you can add more (without using selection_add) or just replace all the suggested task states and replace by your preference.

I have a module that with this change will be viable to use this module.